workflows/hic/hicpro-core.wdl

```diff
@@ -0,0 +1,1103 @@
+        rs_stats: "RS statistics"
+        prefix: "Prefix for the output files"
+    }
+
+    input {
+        Array[File] read1_mapping_stats
+        Array[File] read2_mapping_stats
+        Array[File] valid_pairs_stats
+        Array[File] rs_stats
+        String prefix
+    }
+
+    command <<<
+        set -euo pipefail
+
+        # merge read1 mapping stats
+        mkdir read1
+        ln -s ~{sep(" ", read1_mapping_stats)} read1/
+        python /HiC-Pro_3.0.0/scripts/merge_statfiles.py \
+            -d read1 \
+            -p "*.R1*.mapstat" \
+            -v \
+            > ~{prefix}_R1.mmapstat
+        # merge read2 mapping stats
+        mkdir read2
+        ln -s ~{sep(" ", read2_mapping_stats)} read2/
+        python /HiC-Pro_3.0.0/scripts/merge_statfiles.py \
+            -d read2 \
+            -p "*.R2*.mapstat" \
+            -v \
+            > ~{prefix}_R2.mmapstat
+        # merge pairing stats
+        mkdir pairs
+        ln -s ~{sep(" ", valid_pairs_stats)} pairs/
+        python /HiC-Pro_3.0.0/scripts/merge_statfiles.py \
+            -d pairs \
+            -p "*.pairstat" \
+            -v \
+            > ~{prefix}.mpairstat
+        # merge RS stat
+        mkdir rsstat
+        ln -s ~{sep(" ", rs_stats)} rsstat/
+        python /HiC-Pro_3.0.0/scripts/merge_statfiles.py \
+            -d rsstat \
+            -p "*.RSstat" \
+            -v \
+            > ~{prefix}.mRSstat
+    >>>
+
+    output {
+        File read1_mapping_stats_merged = prefix + "_R1.mmapstat"
+        File read2_mapping_stats_merged = prefix + "_R2.mmapstat"
+        File valid_pairs_stats_merged = prefix + ".mpairstat"
+        File rs_stats_merged = prefix + ".mRSstat"
+    }
+
+    runtime {
+        container: "nservant/hicpro:3.0.0"
+        maxRetries: 1
+    }
+}
+
+task build_raw_maps {
+    meta {
+        description: "Build raw maps from valid pairs for all specified resolutions"
+        outputs: {
+            contact_counts: "Contact counts files for each resolution",
+            contact_counts_bed: "Contact counts for each resolution in BED format",
+        }
+    }
+
+    parameter_meta {
+        chromsizes: "Chromosome sizes file"
+        hic_file: "Hi-C file"
+        bin_sizes: "Bin sizes for each contact counts file"
+        prefix: "Prefix for the output files"
+        allele_specific_snp: "Use allele specific SNP"
+        capture_target: "Capture target file"
+        genome_fragment: "Genome fragment file"
+        matrix_format: "Matrix format"
+    }
+
+    input {
+        File chromsizes
+        File hic_file
+        Array[Int] bin_sizes
+        String prefix
+        File? genome_fragment
+        String matrix_format = "upper"
+        Boolean allele_specific_snp = false
+        Boolean capture_target = false
+    }
+
+    command <<<
+        set -euo pipefail
+
+        for bin_size in ~{sep(" ", bin_sizes)}
+        do
+            cat ~{hic_file} \
```

Comment:
I think `shellcheck` will complain about this. Speaking of, please check `shellcheck` output for _everything_ added in this PR. I love shellcheck 💯